pandas/io/json/_normalize.py

```diff
@@ -226,14 +227,26 @@ def _json_normalize(
     Returns normalized data with columns prefixed with the given string.
     """
 
-    def _pull_field(js: Dict[str, Any], spec: Union[List, str]) -> Iterable:
+    def _pull_field(js: Dict[str, Scalar], spec: Union[List, str]) -> Scalar:
+        """Internal function to pull field"""
         result = js  # type: ignore
         if isinstance(spec, list):
             for field in spec:
-                result = result[field]
+                result = result[field]  # type: ignore
```

Comment:
Can you advise specifically what the error is? I realize we want to get this in for 1.0.2 so not going to block, but I still think this code is suspect (not your fault - just a historical artifact) so I'd hate to suppress a warning about another bug that this fix could be introducing

Reply:
Took a look at this locally; I think if you revert some of the other changes here you won't need the ignore. Suggested separately

Reply:
yeah, having `Any` instead of `Scalar` could fix the issue